### PR TITLE
Move recording snapshots to commit trigger

### PR DESCRIPTION
**Pull request by marianeum**
+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] Component `README.md`
+ [ ] Tests
+ [ ] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

File: .github/workflows/_build.yml [hunk 1]
```diff
@@ -118,11 +118,28 @@ jobs:
           disable-animations: true
           script: ./gradlew :Backpack:connected${{ env.config }}AndroidTest :backpack-compose:connected${{ env.config }}AndroidTest :backpack-common:connected${{ env.config }}AndroidTest
 
+  CheckCommitMessage:
```
Comment (marianeum) [inline .github/workflows/_build.yml:121]: CI checks out a version that merges main into the branch, so need to specifically check out the correct version to get the latest commit message

File: .github/workflows/_build.yml [hunk 2]
```diff
@@ -152,7 +168,8 @@ jobs:
           python-version: '3.x'
           cache: 'pip'
           cache-dependency-path: 'snapshot-requirements.txt'
-      - run: pip3 install -r snapshot-requirements.txt
+      - name: Install Python dependencies
```
Comment (marianeum) [inline .github/workflows/_build.yml:171]: just gave this a name

File: .github/workflows/_build.yml [hunk 3]
```diff
@@ -191,8 +208,12 @@ jobs:
             adb logcat > logcat.log &
             ./gradlew :app:record${{ env.flavour }}${{ env.config }}AndroidTestScreenshotTest -Pandroid.testInstrumentationRunnerArguments.variant=${{ matrix.variant }}
 
-            changedFiles=`git diff --name-only` && echo "::set-output name=CHANGED_FILES::${changedFiles//$'\n'/'%0A'}"
-            ./scripts/check-no-changes.sh
+            changedFiles=`git diff --name-only` && echo "CHANGED_FILES=${changedFiles//$'\n'/'%0A'}" >> $GITHUB_OUTPUT
+
+      - name: Check snapshot changes
```
Comment (marianeum) [inline .github/workflows/_build.yml:213]: this script will fail the build when there's changes - we only want that if we're not updating snapshots, so moved to a separate step

File: .github/workflows/_build.yml [hunk 4]
```diff
@@ -210,17 +231,34 @@ jobs:
           path: logcat.log
           retention-days: 5
 
-      - name: Notify PR of screenshots failure
-        if: ${{ failure() && steps.screenshotTests.outputs.CHANGED_FILES != '' }}
+      - name: Notify PR of screenshots changes
+        if: ${{ (failure() || needs.CheckCommitMessage.outputs.head-commit-message == 'Record snapshots') && steps.screenshotTests.outputs.CHANGED_FILES != '' }}
```
Comment (marianeum) [inline .github/workflows/_build.yml:235]: also add a comment if we're updating snapshots to provide better visibility

File: .github/workflows/_build.yml [hunk 5]
```diff
@@ -210,17 +231,34 @@ jobs:
...
               owner: context.repo.owner,
               repo: context.repo.repo,
-              body: `### '${{ matrix.variant }}' files were changed.
-              If these changes are intentional please run \`./scripts/record_screenshot_tests.sh\` locally and commit the changes.
-
-              >'${{ steps.screenshotTests.outputs.CHANGED_FILES }}'`
+              body: body
             })
+
+      - name: Commit changes
```
Comment (marianeum) [inline .github/workflows/_build.yml:252]: commit & push the changes - need to check out the right branch for this

**Review by frugoman — APPROVED** [review]

**Pull request merged**